src/pages/workspaces/workspace/analysis/modals/CloudEnvironmentModal.js

```diff
@@ -413,7 +415,7 @@ export const CloudEnvironmentModal = ({
   )
 
   const getAzureView = () => Utils.switchCase(viewMode,
-    [runtimeToolLabels.JupyterLab, renderAzureModal],
+    [runtimeToolLabels.JupyterLab, () => renderAzureModal(runtimeToolLabels.JupyterLab)],
```

Comment:
hmm... I don't think it's desirable to pass the tool into the `AzureComputeModalBase`. Seems like its only being used in `renderPersistentDisk`, and the AzureComputeModal itself should decide what to pass into that, not the parent component.

Reply:
Currently it's the approach ComputeModal takes. Additionally, we don't have tool saved in compute or runtime config anywhere, which is where I might expect it to live.

I think this should be taken care of in a follow up ticket, not here.